### PR TITLE
ADD: different DB2 platforms

### PR DESCRIPTION
**Pull request by rPraml**
This will split up in different DB2 platforms.
It is now required to specify which DB2 platform you have. See #2514

### CODE REVIEW

File: .github/workflows/multi-platform-test.yml [hunk 1]
```diff
@@ -43,3 +43,5 @@ jobs:
       run: mvn clean test -Dprops.file=testconfig/ebean-sqlserver17.properties
     - name: mariadb
       run: mvn clean test -Dprops.file=testconfig/ebean-mariadb.properties
+    - name: db2
+      run: mvn clean test -Dprops.file=testconfig/ebean-db2.properties
```
Comment (rPraml) [inline .github/workflows/multi-platform-test.yml:47]: How can I run that workflow?

File: ebean-ddl-generator/src/test/resources/dbmigration/migrationtest/db2/1.1.sql [hunk 2]
```diff
@@ -36,16 +36,16 @@ alter table migtest_e_basic alter column status set default 'A';
 alter table migtest_e_basic alter column status set not null;
 alter table migtest_e_basic add constraint ck_mgtst__bsc_stts check ( status in ('N','A','I','?'));
 alter table migtest_e_basic drop constraint ck_mgtst__b_z543fg;
-alter table migtest_e_basic alter column status2 varchar(127);
+alter table migtest_e_basic alter column status2 set data type varchar(127);
 alter table migtest_e_basic alter column status2 drop default;
-alter table migtest_e_basic alter column status2 set null;
+alter table migtest_e_basic alter column status2 drop not null;
```
Comment (rPraml) [inline ebean-ddl-generator/src/test/resources/dbmigration/migrationtest/db2/1.1.sql:41]: The old statements were wrong. I do not know if they have ever worked.
An `alter table migtest_e_basic alter column status2 set null;` will not work on DB2 11.5 LUW.
@bigjimlefou maybe you can test that on DB2ForI

File: ebean-core/src/main/java/io/ebeaninternal/server/core/DatabasePlatformFactory.java [hunk 3]
```diff
@@ -159,7 +174,7 @@ private DatabasePlatform byDatabaseMeta(DatabaseMetaData metaData, Connection co
     } else if (dbProductName.contains("sqlite")) {
       return new SQLitePlatform();
     } else if (dbProductName.contains("db2")) {
-      return new DB2Platform();
+      throw new IllegalArgumentException("For DB2 please explicitly choose either db2legacy/db2luw/db2zos/db2fori platform. Refer to issue #2514 for details");
```
Comment (rPraml) [inline ebean-core/src/main/java/io/ebeaninternal/server/core/DatabasePlatformFactory.java:177]: @rbygrave Should we make some autodetection here?

File: .github/workflows/multi-platform-test.yml (hunk 1 above)
Comment (rbygrave) [reply .github/workflows/multi-platform-test.yml:47]: Maybe it's a permissions thing ... let me see

File: ebean-core/src/main/java/io/ebeaninternal/server/core/DatabasePlatformFactory.java (hunk 3 above)
Comment (rbygrave) [reply ebean-core/src/main/java/io/ebeaninternal/server/core/DatabasePlatformFactory.java:177]: I am happy for it to be explicit.  Maybe go with this for now and see later on if auto-detection is worthwhile?

**Pull request merged**